dvc/dependency/repo.py

```diff
@@ -94,29 +94,25 @@ def dumpd(self, **kwargs) -> dict[str, Union[str, dict[str, str]]]:
         }
 
     def download(self, to: "Output", jobs: Optional[int] = None):
-        from dvc_data.hashfile.build import build
-        from dvc_data.hashfile.checkout import CheckoutError, checkout
-
-        try:
-            repo = self._make_fs(locked=True).repo
-
-            _, _, obj = build(
-                repo.cache.local,
-                self.fs_path,
-                repo.dvcfs,
-                repo.cache.local.fs.PARAM_CHECKSUM,
-            )
-            checkout(
-                to.fs_path,
-                to.fs,
-                obj,
-                self.repo.cache.local,
-                ignore=None,
-                state=self.repo.state,
-                prompt=confirm,
-            )
-        except (CheckoutError, FileNotFoundError):
-            super().download(to=to, jobs=jobs)
+        from dvc.fs import LocalFileSystem
+
+        files = super().download(to=to, jobs=jobs)
+        if not isinstance(to.fs, LocalFileSystem):
+            return files
+
+        hashes: list[tuple[str, HashInfo, dict[str, Any]]] = []
+        for src_path, dest_path in files:
+            try:
+                hash_info = self.fs.info(src_path)["dvc_info"]["entry"].hash_info
```

Comment:
This may be expensive, as this will also collect `info()` for git-files or local-files, whereas we are only interested for git-tracked files.

Reply:
We could introduce `dvc_only=False|True` in `dvcfs.info()` in the future PRs (if they become a problem).